get_next_line_bonus.c

```diff
@@ -20,15 +20,24 @@ static char	*append_to_save(char **save, int fd)
 
 	buffer = malloc(BUFFER_SIZE + 1);
 	if (!buffer)
+	{
+		free(*save);
+		*save = NULL;
 		return (NULL);
+	}
 	while (true)
 	{
 		read_size = read(fd, buffer, BUFFER_SIZE);
 		if (read_size <= 0)
 			break ;
 		buffer[read_size] = '\0';
-		*save = ft_strjoin_with_free(*save, buffer, 0b10);
-		nl_ptr = ft_strchr(*save, '\n');
+		*save = gnl_strjoin_with_free(save, &buffer, 0b10);
+		if (!*save)
+		{
+			free(buffer);
+			return (NULL);
+		}
```

Comment:
I think this if statement is unnecessary.
It seems necessary for performance improvement, but whether this if statement exists or not, the TLE issue will not be resolved. It would be better to remove it for issue#5 and norm.

Reply:
Confirmed. You are right. It is an unneccessary if statement.